### PR TITLE
- Get pango 2D depiction to work with cairocffi

### PR DESCRIPTION
**Pull request by ptosco**
I have noticed that Pango/PangoCairo text handling (superscript/subscript) does not work with `cairocffi`. This PR gets it to work again, while keeping compatibility with the old pycairo/pango/pangocairo Python 2 modules. Tested under all platforms with both Python 2 and 3.

I am aware that the Python 2D depiction will become soon obsolete once the C++ depiction module becomes feature-complete, but I think this can be useful for the time being.

### CODE REVIEW

File: rdkit/Chem/Draw/cairoCanvas.py [hunk 1]
```diff
@@ -10,44 +10,120 @@
...
+        void pango_font_description_free (PangoFontDescription *desc);
+        void pango_font_description_set_family (PangoFontDescription *desc,
+            const char *family);
+        void pango_font_description_set_size (PangoFontDescription *desc,
+            int size);
+        void pango_layout_set_font_description (PangoLayout *layout,
+            const PangoFontDescription *desc);
+    ''')
+    if (platform.system() == 'Windows'):
+      defaultLibs = {
```
Comment (greglandrum) [inline rdkit/Chem/Draw/cairoCanvas.py]: Is there any way to do this without hard-coding library versions like this?

Comment (ptosco) [inline rdkit/Chem/Draw/cairoCanvas.py]: Hi Greg. Are you referring to the library names? Unfortunately the `ctypes.util` module is not the smartest around. On Linux it requires at a minimum "pango-1", "pangocairo-1" and "gobject-2". On OS X, you also need the ".0". On Windows, as there is no general accepted naming convention, you need the full name. This is why, along with the defaults, I also implemented the possibility to override the defaults with environment variables `RDK_PANGO_LIB`, `RDK_PANGOCAIRO_LIB`, `RDK_GOBJECT_LIB`, such that if `ctypes.util` fails to find the library, you may provide a different library name for each of them.

Comment (greglandrum) [inline rdkit/Chem/Draw/cairoCanvas.py]: Yeah, I was asking about the library names. If that's the only way to do it, then I guess we have to live with it.

File: rdkit/Chem/Draw/cairoCanvas.py [hunk 2]
```diff
@@ -10,44 +10,120 @@
...
+      else:
+        libName = defaultLibs[libType + '_default_lib']
+      libPath = ctypes.util.find_library(libName)
+      exec(libType + ' = None')
+      if (libPath):
+        try:
+          exec(libType + ' = ffi.dlopen("' \
+            + libPath.replace('\\', '\\\\') + '")')
+        except:
+          pass
```
Comment (bp-kelley) [inline rdkit/Chem/Draw/cairoCanvas.py]: If the RDK_ environments are set and the libs aren't found, I think it it reasonable to raise an exception along the lines of

```
ImportError: RDK_PANGO_LIB set to /usr/lib/libpango..., but PANGO library cannot be loaded.
```

**Comment by ptosco** [comment]
@bp-kelley : done. I have noticed that the exception is caught upstream in `rdkit/Chem/Draw/__init__.py`, but it isn't if `RDKIT_CANVAS` is explicitly set to `cairo`, which is coherent.

**Comment by greglandrum** [comment]
@ptosco I just realized that I still haven't merged this.
I'm happy to do so, but unfortunately there are now conflicts thanks to the python re-formatting that was recently done. Do you want me to resolve these?

**Comment by ptosco** [comment]
Hi Greg, no worries, I can resolve the conflicts for you tonight or tomorrow morning.

**Comment by ptosco** [comment]
@greglandrum Hi Greg, you should be able to merge now,

**Comment by greglandrum** [comment]
Thanks Paolo!

**Pull request merged**